network/transport/v2/protocol.go

```diff
@@ -142,19 +140,27 @@ func (p protocol) UnwrapMessage(envelope interface{}) interface{} {
 }
 
 func (p *protocol) Configure(_ transport.PeerID) error {
-	dbFile := path.Join(p.config.Datadir, "network", "payload_jobs.db")
-	if err := os.MkdirAll(filepath.Dir(dbFile), os.ModePerm); err != nil {
-		return fmt.Errorf("unable to setup database: %w", err)
-	}
-
-	db, err := bbolt.Open(dbFile, 0600, bbolt.DefaultOptions)
+	nodeDID, err := p.nodeDIDResolver.Resolve()
 	if err != nil {
-		return fmt.Errorf("unable to create BBolt database: %w", err)
+		log.Logger().WithError(err).Error("Failed to resolve node DID")
 	}
 
-	p.payloadScheduler, err = NewPayloadScheduler(db, p.config.PayloadRetryDelay, p.handlePrivateTxRetry)
-	if err != nil {
-		return fmt.Errorf("failed to setup payload scheduler: %w", err)
+	if nodeDID.Empty() {
+		log.Logger().Warn("Not starting the payload scheduler as node DID is not set")
+	} else {
+		store, err := p.storageProvider.GetKVStore("data", storage.PersistentStorageClass)
```

Comment:
Wouldn't it be better to just pass the store into protocol in the `New()` function? Probably easier testing, and at the place it's being called, it should be there, so easy fix?

`StorageProvider` can return an existing KVStore, but feels a bit weird and error prone to request the `data` KV store here?